assignments/this.js

```diff
@@ -13,14 +13,42 @@
 
 // code example for Window Binding
 
-// Principle 2
+console.log(this);
 
+// Principle 2
 // code example for Implicit Binding
 
-// Principle 3
+const person = {
+name: "Rodean",
+age: 31,
+school: "Lambda",
+speak: function(){
+    return (`Hi my name is ${this.name} how are you doing?`);
+    }
+}
+console.log(person.speak());
 
+// Principle 3
 // code example for New Binding
 
+const People = function(name, age, school){
+    this.name = name;
+    this.age = age;
+    this.school = school;
+    this.talk = function(){
+        return `My name is ${this.name} I am ${this.age} and I attend ${this.school}`;
+    };
+};
+
+const robert = new People("Robert", 22, "FAU");
+const ashley = new People("Ashley", 23, "FSU");
+console.log(robert.talk())
+
 // Principle 4
+// code example for Explicit Binding
+
+function intro(person){
+    return (`Hello everyone my name is ${this.name} and I am a student at ${this.school}`);
```

Comment:
the code example you have for principle 4 is implicit binding, 
here is an example of explicit binding
```
function Van(attributes) {
  Car.call(this, attributes);
  this.type = attributes.type;
}

const aeroStar = new Van({
  name: "AeroStar",
  maxSpeed: "really slow",
  size: "big",
  color: "dirty black",
  generation: "old",
  nickName: "DeathStar",
  type: "ugly car"
});

console.log(aeroStar, "\n principle 4");
```

as you can see I am using `.call` to explicit say what is `this`, and `this is referring to `Car`.